geode-for-redis/src/test/java/org/apache/geode/redis/internal/data/delta/ReplaceByteArrayAtOffsetDeltaUnitTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.redis.internal.data.delta;
+
+import static org.apache.geode.redis.internal.data.delta.DeltaType.REPLACE_BYTE_ARRAY_AT_OFFSET;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+import junitparams.Parameters;
+import junitparams.naming.TestCaseName;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.apache.geode.redis.internal.data.RedisData;
+import org.apache.geode.redis.internal.data.RedisList;
+import org.apache.geode.redis.internal.data.RedisString;
+import org.apache.geode.test.junit.runners.GeodeParamsRunner;
+
+@RunWith(GeodeParamsRunner.class)
+public class ReplaceByteArrayAtOffsetDeltaUnitTest extends AbstractRedisDeltaUnitTest {
+  private final String payload = "something amazing I guess";
+
+  @Test
+  public void testReplaceByteArrayAtOffsetDeltaForRedisString() throws Exception {
+    DataInputStream dis = getDataInputStream(3);
+    String original = "0123456789";
+    RedisString redisString = new RedisString(original.getBytes());
+
+    redisString.fromDelta(dis);
+
+    assertThat(new String(redisString.get())).isEqualTo(original.substring(0, 3) + payload);
```

Comment:
It might be worth adding another test for this delta with Strings where the payload is small, since we have different logic in the `applyReplaceByteArrayAtOffsetDelta()` method if we're just replacing a subset of the String rather than extending the whole thing. It could look something like this:
```
  @Test
  public void testReplaceByteArrayAtOffsetDelta_forRedisString_withSmallByteArray() throws Exception {
    String payload = "small";
    DataInputStream dis = getDataInputStream(3, payload);
    String original = "0123456789";
    RedisString redisString = new RedisString(original.getBytes());

    redisString.fromDelta(dis);

    assertThat(new String(redisString.get())).isEqualTo("012small89");
  }
```
and would require making `getDataInputStream()` take a String argument to allow the payload to be different in different tests.

Reply:
Updated with distinctions between small and large payloads.